circuits/core/src/aleo.rs

```diff
@@ -0,0 +1,272 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
```

Comment:
Note: This file is very much still a WIP. You are welcome to skim it, it will continue to be refactored in subsequent PRs.